xarray/tests/test_groupby.py

```diff
@@ -94,6 +94,18 @@ def func(arg1, arg2, arg3=0):
     assert_identical(expected, actual)
 
 
+@pytest.mark.xfail
+def test_da_groupby_single_value_per_dim():
+
+    array = xr.DataArray([[1, 1, 1], [2, 2, 2]],
+                         [('x', [1, 2]), ('y', [0, 1, 2])])
+
+    # This raises an error.
+    # I think the issue is that gr._group_indices is [0, 1]
+    # instead of [[0,], [1,]]
+    array.groupby('x').mean(dim='x')
```

Comment:
Don't you just want `array.groupby('x').mean()`? You're grouping over dimension 'x' and then attempting to take a mean along `x` which isn't a meaningful operation.

Reply:
In a case where you have multiple realization of a variable for the same coordinate, doesn't it make sense to compute the mean along the same dimension you grouped on ? I've added a few tests where we group by month, then compute the quantile over time (same dimension as grouping).

Reply:
Yeah but when you group by `x` in this example, each group does not have an `x` dimension. `groupby_bins` on `x` would be meaningful.

We should probably raise a more informative error.

Reply:
Should I remove this test from the PR ?

Reply:
I think so. You could open an issue for further discussion.